File: Second Semester/Data Structures and Algorithms I/Practice 11/calendarizacion.py
#Actividad 2 - Practica 11
#Santiago Díaz Edwin Jaret
#Estructura de Datos y Algoritmos

def activities(s, f, n):
    print("Selected Activities are:")
    i = 0
    print("A" + str(i+1))
    for j in range(1, n):
        if s[j] >= f[i]:
            print("A" + str(j+1)) # Si la hora de finalización es igual o mayor a la hora de inicio
            i = j # La hora de inicio ahora es la hora final de la ultima actividad

if __name__ == '__main__':
    s = [1, 2, 3, 2, 4, 5, 6, 8, 7] #Horario de inicio de actividades
    f = [4, 5, 6, 8, 6, 7, 7, 12, 9] #Horario de fin de actividades
    #Esto lo hace diferente al programa escrito en C
    n = len(s)
    activities(s, f, n)

File: Second Semester/Data Structures and Algorithms I/Practice 11/calendarizacion.c
//Actividad 2 - Practica 11
//Santiago Díaz Edwin Jaret
//Estructura de Datos y Algoritmos

#include <stdio.h>

void activities(int s[], int f[], int n) {
    int i, j;
    printf ("Selected Activities are:\n");
    i = 0;
    printf("A%d ", i+1);
    for (j = 1; j < n; j++) {
        if (s[j] >= f[i]){ //Si la hora de finalización es igual o mayor a la hora de inicio
            printf ("A%d ", j+1);
            i = j; //La hora de inicio ahora es la hora final de la ultima actividad
        }
    }
}

void main(){
    int s[] = {1, 2, 3, 2, 4, 5, 6, 8, 7}; //horarios inicio de actividades
    int f[] = {4, 5, 6, 8, 6, 7, 7, 12, 9}; //horario fin de actividades
    int n = sizeof(s)/sizeof(s[0]); //Tamano de la lista
    activities(s, f, n); 
    getchar();
}